testsuite/MDAnalysisTests/analysis/test_pca.py

```diff
@@ -145,6 +145,101 @@ def test_transform_universe():
     pca_test.transform(u2)
 
 
+def test_project_no_pca_run(u, pca):
+    pca_class = PCA(u, select=SELECTION)
+    with pytest.raises(ValueError):
+        func = pca_class.project_single_frame()
+
+
+def test_project_none_anchor(u, pca):
+    group = u.select_atoms('resnum 1')
+    with pytest.raises(ValueError):
+        func = pca.project_single_frame(0, group=group, anchor=None)
+
+
+def test_project_more_anchor(u, pca):
+    group = u.select_atoms('resnum 1')
+    with pytest.raises(ValueError):
+        project = pca.project_single_frame(0, group=group, anchor='backbone')
+
+
+def test_project_less_anchor(u, pca):
+    group = u.select_atoms('all')
+    with pytest.raises(ValueError):
+        project = pca.project_single_frame(0, group=group, anchor='name CB')
+
+
+def test_project_improper_anchor(u):
+    pca = PCA(u, select='name CA').run()
+    group = u.select_atoms('all')
+    with pytest.raises(ValueError):
+        project = pca.project_single_frame(0, group=group, anchor='name N')
+
+
+def test_project_compare_projections():
+    u = mda.Universe(PSF, DCD)
+    u_copy = u.copy()
+    pca = PCA(u, select=SELECTION).run()
+    project0 = pca.project_single_frame(0)
+    project1 = pca.project_single_frame(1)
+
+    u.trajectory[0]
+    coord0 = project0(u.trajectory.ts).positions.copy()
+    u.trajectory[0]
+    coord1 = project1(u.trajectory.ts).positions.copy()
+
+    with pytest.raises(AssertionError):
+        assert_almost_equal(coord0, coord1, decimal=5)
+
+
+def test_project_reconstruct_whole():
+    u = mda.Universe(PSF, DCD)
+    u_copy = u.copy()
+    pca = PCA(u, select=SELECTION).run()
+    project = pca.project_single_frame()
+
+    coord_reconstructed = project(u.trajectory.ts).positions.copy()
+    coord_copy = u_copy.trajectory.ts.positions.copy()
+    assert_almost_equal(coord_reconstructed, coord_copy, decimal=5)
+
+
+def test_project_twice_projection():
+    u = mda.Universe(PSF, DCD)
+    pca = PCA(u, select=SELECTION).run()
+
+    project0 = pca.project_single_frame(0)
+    project1 = pca.project_single_frame(1)
+
+    u.trajectory[0]
+    coord0 = project0(u.trajectory.ts).positions.copy()
+    coord00 = project0(u.trajectory.ts).positions.copy()
+    assert_almost_equal(coord0, coord00, decimal=5)
+
+
+@pytest.mark.parametrize('distance_sel, case',
+                         (('resnum 1 and name CA CB', 1),
+                          ('resnum 1 and name N CA CB', 2)))
+def test_projet_extrapolate_translation(distance_sel, case):
+    u = mda.Universe(PSF, DCD)
+    pca = PCA(u, select='backbone').run()
+    group = u.select_atoms(distance_sel)
+    project = pca.project_single_frame(0, group=group,
+                                       anchor='resnum 1 and name CA')
+
+    distances_original = (
+        mda.lib.distances.self_distance_array(group.positions.copy())
```

Comment:
Done. All the copy() are removed, except where the function is applied twice in succession.

Reply:
Furthermore, `ATomGroup.positions` is already a copy!

Reply:
Is it? This may require a further investigation of the `positions` method. I have a `.copy()` method in test_project_twice_projection (line 230) and I absolutely need it. (Copy in line 231 can be removed though. I'm on it.)

I also compared the actual numbers. The test works as intended with`project_first(u_fresh.trajectory.ts).positions.copy()`, but not with `project_first(u_fresh.trajectory.ts).positions`